code/go/0chain.net/smartcontract/storagesc/allocation.go

```
@@ -96,24 +93,51 @@ type newAllocationRequest struct {
 	ThirdPartyExtendable bool       `json:"third_party_extendable"`
 	FileOptionsChanged   bool       `json:"file_options_changed"`
 	FileOptions          uint16     `json:"file_options"`
+
+	IsEnterprise bool `json:"is_enterprise"`
 }
 
 // storageAllocation from the request
-func (nar *newAllocationRequest) storageAllocation(conf *Config, now common.Timestamp) (sa *StorageAllocation) {
-	sa = new(StorageAllocation)
-	sa.DataShards = nar.DataShards
-	sa.ParityShards = nar.ParityShards
-	sa.Size = nar.Size
-	sa.Expiration = common.Timestamp(common.ToTime(now).Add(conf.TimeUnit).Unix())
-	sa.Owner = nar.Owner
-	sa.OwnerPublicKey = nar.OwnerPublicKey
-	sa.PreferredBlobbers = nar.Blobbers
-	sa.ReadPriceRange = nar.ReadPriceRange
-	sa.WritePriceRange = nar.WritePriceRange
-	sa.ThirdPartyExtendable = nar.ThirdPartyExtendable
-	sa.FileOptions = nar.FileOptions
+func (nar *newAllocationRequest) storageAllocation(balances chainstate.StateContextI, conf *Config, now common.Timestamp) *StorageAllocation {
+	sa := &StorageAllocation{}
+
+	_ = chainstate.WithActivation(balances, "electra", func() error {
```

We should not ignore the error, it can be `node not found` error. In that case, we need to return the error and throw it up and stop the txn execution.

---

Here we are creating new structure. Not sure node  not found is relevant here.

---

No matter what we are doing in the functions, the `node not found` error should always be concerned and returned to upper level. We should never ignore the error from `WithActivation`. That means When the error happens, we need to stop.